assets/js/modules/analytics-4/components/setup/SetupEnhancedMeasurementSwitch.js

```diff
@@ -109,30 +113,15 @@ export default function SetupEnhancedMeasurementSwitch() {
 		);
 	} );
 
-	const isEnhancedMeasurementEnabled = useSelect( ( select ) =>
-		select( CORE_FORMS ).getValue(
-			ENHANCED_MEASUREMENT_FORM,
-			ENHANCED_MEASUREMENT_ENABLED
-		)
-	);
-
 	const { setValues } = useDispatch( CORE_FORMS );
 
-	// If `isEnhancedMeasurementEnabled` is already defined in the first render, it means we're rendering this component
-	// after the user has actively selected the setting, in which case we don't want to override the setting to `true` in
-	// the `useEffect()` unless the account, property, or web data stream selection is subsequently changed.
-	const skipEffect = useRef( isEnhancedMeasurementEnabled !== undefined );
-
-	useEffect( () => {
-		if ( skipEffect.current ) {
-			skipEffect.current = false;
-			return;
+	useMount( () => {
+		if ( ! autoSubmit ) {
```

Comment:
I included this one instead, so by default - unless returning from OAuth, we push the EM to `on`. Then user interactivity with actual dropdown would reset EM value next time, to ensure the user selected value is respected

Reply:
Can we identify this without checking `autoSubmit`? Ideally we wouldn't reference that outside of the main component where it happens.

Reply:
I couldn't identify more reliable way to catch the redirection moment. As any other value can be used either way. If I try to check for `ACCOUNT_CREATE` for example, then it would not be switched to default on mount if the account does not have a property and creating a new one is preselected by default, or on page refresh, etc. We do use `autoSubmit` check in several different components to confirm the OAuth redirect